src/safeposix/syscalls/net_calls.rs

```diff
@@ -1234,4 +1233,77 @@ impl Cage {
+            if let Socket(socketobj) = &*filedesc {
+                //if the socket is not connected, then we should return an error
+                if socketobj.remoteaddr == None {
+                    return syscall_error(Errno::ENOTCONN, "getpeername", "the socket is not connected");
+                }
+                
+                //all of the checks that we had passed if we are here
+                *ret_addr = socketobj.remoteaddr.unwrap();
+                return 0;
+
+            } else {
+                return syscall_error(Errno::ENOTSOCK, "getpeername", "the provided file is not a socket");
+            }
+        } else {
+            return syscall_error(Errno::EBADF, "getpeername", "the provided file descriptor is not valid");
+        }
+    }
+
+    pub fn getsockname_syscall(self, fd: i32, ret_addr: &mut interface::GenSockaddr) -> i32 {
+        let fdtable = self.filedescriptortable.read().unwrap();
+
+        if let Some(wrappedfd) = fdtable.get(&fd) {
+            let filedesc = wrappedfd.read().unwrap();
+            if let Socket(socketobj) = &*filedesc {
+                if socketobj.localaddr == None {
+                    
+                    //sets the address to 0.0.0.0 if the address is not initialized yet
+                    match ret_addr {
+                        interface::GenSockaddr::V4(_) => {
+                            let addr = interface::GenIpaddr::V4(interface::V4Addr::default());
+                            ret_addr.set_port(0);
+                            ret_addr.set_addr(addr);
+                        }
+        
+                        interface::GenSockaddr::V6(_) => {
+                            let addr = interface::GenIpaddr::V6(interface::V6Addr::default());
+                            ret_addr.set_port(0);
+                            ret_addr.set_addr(addr);
+                        }
+                    }
+                    return 0;
+                }
+                
+                //if the socket is not none, then return the socket
+                *ret_addr = socketobj.localaddr.unwrap();
+                return 0;
+
+            } else {
+                return syscall_error(Errno::ENOTSOCK, "getsockname", "the provided file is not a socket");
+            }
+        } else {
+            return syscall_error(Errno::EBADF, "getsockname", "the provided file descriptor is not valid");
+        }
+    }
+
+    //we only return the default host name because we do not allow for the user to change the host name right now
+    pub fn gethostname(self, length: usize, mut address_ptr: &[u8]) -> i32 {
+        if length < 0 {
+            return syscall_error(Errno::EINVAL, "gethostname", "invalid argument");
+        }
+        if DEFAULT_HOSTNAME.chars().count() > length {
+            address_ptr = &DEFAULT_HOSTNAME[..length].as_bytes();
+        } else {
+            address_ptr = &*DEFAULT_HOSTNAME.as_bytes();
+        }
+        return 0;
+    }
```

Comment:
This function fills in the data incorrectly. Rather than a mutable slice handle which we can change from within the function, we want to change the data that's pointed to by the slice.
So: `pub fn gethostname(self, length: usize, address_ptr: &mut [u8]) -> i32 {`
And then to change the data pointed to by the address_ptr, use the copy_from_slice function, which you can see here: https://doc.rust-lang.org/std/primitive.slice.html#method.copy_from_slice.

Reply:
I think that I fixed it. If I didn't, please let me know.